furbaby/api/views.py

```diff
@@ -177,3 +187,170 @@ def password_reset_token_created(sender, instance, reset_password_token, *args,
+    def update_location_record(self, request):
+        try:
+            updated_fileds = []
+            location_id = request.data["id"]
+            location = Locations.objects.get(id=location_id)
+            if "address" in request.data:
+                location.address = request.data["address"]
+                updated_fileds.append("address")
+            if "city" in request.data:
+                location.city = request.data["city"]
+                updated_fileds.append("city")
+            if "country" in request.data:
+                location.country = request.data["country"]
+                updated_fileds.append("country")
+            if "zipcode" in request.data:
+                location.zipcode = request.data["zipcode"]
+                updated_fileds.append("zipcode")
+            if "default_location" in request.data:
+                if request.data["default_location"] == True:
+                    # unset all other locations as default
+                    Locations.objects.filter(user_id=request.user.id).update(default_location=False)
+                location.default_location = request.data["default_location"]
+                updated_fileds.append("default_location")
+
+            # location.save()
+            location.save(update_fields=updated_fileds)
+
+            return json_response(
+                self.get_location_record(location),
+                status.HTTP_200_OK,
+                safe=False,
+                include_data=False,
+            )
+        except Locations.DoesNotExist:
+            return json_response(
+                data={
+                    "error": "location not found for user",
+                    "location id": location_id,
+                    "user id": request.user.id,
+                },
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        except Exception as e:
+            return json_response(
+                data={
+                    "error": "something went wrong while updating the location record",
+                    "error message": str(e),
+                    "location id": location_id,
+                    "user id": request.user.id,
+                },
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+    def delete_location_record(self, request):
+        try:
+            location_id = request.data["id"]
+            location = Locations.objects.get(id=location_id)
+            location.delete()
+            return json_response(
+                {"message": "location deleted successfully"},
+                status.HTTP_200_OK,
+            )
+        except Locations.DoesNotExist:
+            return json_response(
+                data={
+                    "error": "location not found for user",
+                    "location id": location_id,
+                    "user id": request.user.id,
+                },
+                status=status.HTTP_404_NOT_FOUND,
+            )
```

Comment:
nit-pick:

what you've done here is completely fine, but I was thinking that if we're going to check it every time, it can become a little redundant. What do you think of making it one common test in the main handler and we assume that the location is always present when you're in this function?

Reply:
Shall we take this up in the next PR?